### PR TITLE
Bugfix: Disabled DateBox does not open dialog

### PR DESCRIPTION
**Pull request by schube**
Currently, if a DateBox is disabled, it still opens the dialog if clicked.

This commit fixes this behaviour.

### CODE REVIEW

File: src/main/java/org/dominokit/domino/ui/datepicker/DateBox.java [hunk 1]
```diff
@@ -63,7 +63,10 @@ protected HTMLInputElement createElement(String type, String placeholder) {
     private void init() {
         this.pattern = this.datePicker.getDateTimeFormatInfo().dateFormatFull();
         this.datePicker.addDateSelectionHandler((date, dateTimeFormatInfo) -> setStringValue(date, dateTimeFormatInfo));
-        this.modalListener = evt -> modal.open();
+        this.modalListener = evt -> {
+        		if(!super.isDisabled())
+        			modal.open();
```
Comment (niloc132) [inline src/main/java/org/dominokit/domino/ui/datepicker/DateBox.java]: `if` should have `{}`s, and a space between `if` and `(`

**Comment by niloc132** [comment]
It looks like this patch is using tabs instead of spaces, which appears to be the convention for this project.

File: src/main/java/org/dominokit/domino/ui/forms/TextFormElement.java [hunk 2]
```diff
@@ -25,7 +25,16 @@ private void addFocusListeners() {
     public HTMLDivElement asElement() {
         return formGroup;
     }
+    
+	public boolean isDisabled() {
+		String disabled = getInputElement().getAttribute("disabled");
+		if(null == disabled) {
+			return false;
```
Comment (niloc132) [inline src/main/java/org/dominokit/domino/ui/forms/TextFormElement.java]: Looks like tabs are still being used here

File: src/main/java/org/dominokit/domino/ui/datepicker/DateBox.java [hunk 3]
```diff
@@ -63,7 +63,11 @@ protected HTMLInputElement createElement(String type, String placeholder) {
     private void init() {
         this.pattern = this.datePicker.getDateTimeFormatInfo().dateFormatFull();
         this.datePicker.addDateSelectionHandler((date, dateTimeFormatInfo) -> setStringValue(date, dateTimeFormatInfo));
-        this.modalListener = evt -> modal.open();
+        this.modalListener = evt -> {
+        	  if(!super.isDisabled()) {
+        	    modal.open();
```
Comment (niloc132) [inline src/main/java/org/dominokit/domino/ui/datepicker/DateBox.java]: Sorry to nag, but while this switched to spaces, you are using 2 spaces here, but the convention is 4.

Also, space between `if` and `(` (also in the other file)

File: src/main/java/org/dominokit/domino/ui/forms/TextFormElement.java [hunk 4]
```diff
@@ -25,7 +25,16 @@ private void addFocusListeners() {
...
+		}
+		return true;
```
Comment (niloc132) [inline src/main/java/org/dominokit/domino/ui/forms/TextFormElement.java]: This whole if/return section could be replaced with `return disabled != null;` (or `null != disabled` if you prefer).

**Comment by vegegoku** [comment]
@rjeeb is working on adding more features to the fields, see https://github.com/DominoKit/domino-ui/issues/11
the code is being enhanced so this one will be fixed when he is done. he is supposed to be in finalization stage by now.

**Comment by rjeeb** [comment]
This bug is fixed by this feature #11

**Comment by schube** [comment]
i think it is not fixed. i can click on a disabled datebox and the dialog appears

**Comment by vegegoku** [comment]
should be fixed now

**Pull request closed without merging**